lightningd/plugin.c

```diff
@@ -615,6 +617,8 @@ char *plugin_opt_set(const char *arg, struct plugin_opt *popt)
 		} else
 			return tal_fmt(tmpctx, "%s does not parse as type %s",
 				       popt->value->as_str, popt->type);
+		tal_free(popt->value->as_str);
```

Comment:
To remove this code here, is a good choice to insert another if and check if the propriety is a string?